src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -81,20 +81,21 @@ public override void Initialize(AnalysisContext context)
 
                 var equalityComparerMethods = GetEqualityComparerMethodsToCheck(compilation);
                 var systemHashCode = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemHashCode);
+                var iEqualityComparer = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemCollectionsGenericIEqualityComparer1);
 
                 context.RegisterOperationAction(
-                    context => HandleInvocationOperation(in context, symbolType, symbolEqualityComparerType, equalityComparerMethods, systemHashCode),
+                    context => HandleInvocationOperation(in context, symbolType, symbolEqualityComparerType, equalityComparerMethods, systemHashCode, iEqualityComparer),
                     OperationKind.Invocation);
 
-                if (symbolEqualityComparerType != null)
+                if (symbolEqualityComparerType is not null && iEqualityComparer is not null)
```

Comment:
I'm still keeping the `symbolEqualityComparerType` null check so that we don't report diagnostics when `SymbolEqualityComparer` doesn't exist, even if `IEqualityComparer` can be implemented.

Reply:
As you are not using the type symbol per say, I would change the first part of the if to use `UseSymbolEqualityComparer` helper.